cli/internal/helm/client.go

```diff
@@ -9,34 +9,33 @@ package helm
 import (
 	"context"
 	"fmt"
+	"os"
 	"strings"
 	"time"
 
 	"github.com/edgelesssys/constellation/v2/internal/config"
 	"github.com/edgelesssys/constellation/v2/internal/constants"
 	"github.com/edgelesssys/constellation/v2/internal/deploy/helm"
-	"github.com/pkg/errors"
+	"github.com/spf13/afero"
 	"golang.org/x/mod/semver"
 	"helm.sh/helm/v3/pkg/action"
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/cli"
-	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
-	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/serializer"
-	"k8s.io/client-go/kubernetes/scheme"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 // Client handles interaction with helm.
 type Client struct {
-	config    *action.Configuration
-	crdClient *apiextensionsclient.Clientset
-	log       debugLog
+	config  *action.Configuration
+	kubectl crdClient
+	fs      afero.Fs
+	log     debugLog
 }
```

Comment:
Any specific reason for not using `file.Handler` for file interactions in the `Client`?
As far as I see, this code only makes use of `MkdirAll` and `WriteFile`, which are both supported by `file.Handler`.

Reply:
No, wasn't aware of it. Changed.